packages/z2s/src/compiler.ts

```diff
@@ -596,20 +596,29 @@ export class Compiler {
         this.#nameMapper.columnName(table, column)
       ];
     const serverType = serverColumnSchema.type;
-    if (
-      !serverColumnSchema.isEnum &&
-      (serverType === 'date' ||
+    if (!serverColumnSchema.isEnum) {
+      if (
+        serverType === 'date' ||
         serverType === 'timestamp' ||
+        serverType === 'timestamp without time zone'
+      ) {
+        return sql`EXTRACT(EPOCH FROM ${sql.ident(
+          table,
+        )}.${this.#mapColumnNoAlias(
+          table,
+          column,
+        )}::timestamp AT TIME ZONE 'UTC') * 1000 as ${sql.ident(column)}`;
```

Comment:
Should be:

```suggestion
        return sql`EXTRACT(EPOCH FROM ${sql.ident(
          table,
        )}.${this.#mapColumnNoAlias(
          table,
          column,
        )}) * 1000 as ${sql.ident(column)}`;
```

round-trips correctly:

```
user@127.0.0.1:postgres> CREATE TABLE bar (a timestamp without time zone);
CREATE TABLE
Time: 0.028s
user@127.0.0.1:postgres> INSERT INTO bar VALUES(to_timestamp(1744392199) AT TIME ZONE 'UTC');
INSERT 0 1
Time: 0.011s
user@127.0.0.1:postgres> SELECT * FROM bar;
+---------------------+
| a                   |
|---------------------|
| 2025-04-11 17:23:19 |
+---------------------+
SELECT 1
Time: 0.004s
user@127.0.0.1:postgres> SELECT EXTRACT(EPOCH FROM "a") FROM bar;
+-------------------+
| extract           |
|-------------------|
| 1744392199.000000 |
+-------------------+
SELECT 1
Time: 0.005s
user@127.0.0.1:postgres>
```

The reason is that `EXTRACT(EPOCH FROM` is already returning an epoch time which is at UTC.

Reply:
ah yeh so this one the `::timestamp` cast was stripping the timezone, so then it was being treated as local timezone and converted to UTC, and the epoch taken from that /0\

Reply:
wait no there was no timezone, so that converted it to a `timestamptz`, interpreting as a timestamp in the user local timezone 🤯